starter-code/src/functions-and-arrays.js

```diff
@@ -1,93 +1,454 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(x, y) {
+  return Math.max(x, y);
+}
+
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+
+function findLongestWord(arr) {
+  let longest = "";
+  if (arr.length === 0) {
+    return null;
+  } else {
+    for (i = 0; i < arr.length; i++) {
+      if (arr[i].length > longest.length) {
+        longest = arr[i];
+      }
+    }
+  }
+  return longest;
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(arr1) {
+  let sum = 0;
+  for (let i = 0; i < arr1.length; i++) {
+    sum += arr1[i];
+  }
+  return sum;
+}
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(numArr) {
+  let sum = 0;
+  if (numArr.length === 0) {
+    return null;
+  } else
+    for (let i = 0; i < numArr.length; i++) {
+      sum += numArr[i];
+    }
+  return sum / numArr.length;
+}
+
 // Level 2: Array of strings
 const wordsArr = [
-  'seat',
-  'correspond',
-  'linen',
-  'motif',
-  'hole',
-  'smell',
-  'smart',
-  'chaos',
-  'fuel',
-  'palace'
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
 ];
 
+function averageWordLength(strArray) {
+  let totalWordLength = 0;
+  if (strArray.length === 0) {
+    return null;
+  }
+  for (let i = 0; i < strArray.length; i++) {
+    totalWordLength += strArray[i].length;
+  }
+  return totalWordLength / strArray.length;
+}
+
 // Iteration #5: Unique arrays
 const wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+  "crab",
+  "poison",
+  "contagious",
+  "simple",
+  "bring",
+  "sharp",
+  "playground",
+  "poison",
+  "communion",
+  "simple",
+  "bring"
 ];
 
+function uniquifyArray(arr2) {
+  let newArray = [];
+  for (let i = 0; i < arr2.length; i++) {
+    if (newArray.indexOf(arr2[i]) === -1) {
+      newArray.push(arr2[i]);
+    }
+  }
+  return newArray;
+}
+
 // Iteration #6: Find elements
 const wordsFind = [
-  'machine',
-  'subset',
-  'trouble',
-  'starting',
-  'matter',
-  'eating',
-  'truth',
-  'disobedience'
+  "machine",
+  "subset",
+  "trouble",
+  "starting",
+  "matter",
+  "eating",
+  "truth",
+  "disobedience"
 ];
 
+function doesWordExist(arr3, str2) {
```

Comment:
why arr3 and str2? Maybe it's better to simply name it array, string or just arr, str